cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -165,54 +174,77 @@ func (m *azureCache) fetchAzureResources() error {
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
-	switch m.vmType {
-	case vmTypeVMSS:
-		// List all VMSS in the RG.
-		vmssResult, err := m.fetchScaleSets()
-		if err == nil {
-			m.scaleSets = vmssResult
-		} else {
-			return err
-		}
-	case vmTypeStandard:
-		// List all VMs in the RG.
-		vmResult, err := m.fetchVirtualMachines()
-		if err == nil {
-			m.virtualMachines = vmResult
-		} else {
-			return err
-		}
+	// fetch all the resources since CAS may be operating on mixed nodepools
+	// including both VMSS and VMs pools
```

Comment:
A little concerned about implications (including performance) of doing this always, unconditionally; something to watch out for during testing.